cypress/integration/common/deal-wizard.ts

```diff
@@ -0,0 +1,91 @@
+import { Then, When } from "@badeball/cypress-cucumber-preprocessor/methods";
+
+Then("I am presented the option to choose a partner", () => {
+  cy.url().should("match", /(initiate\/token-swap$)/)
+})
+
+When("I go to previous step", () => {
+  cy.get("[data-test='wizard-previous-button']").click();
+})
```

Comment:
I'm thinking, that we could also verify the url here

1. get current url
2. click button
3. check, that url changed

Reply:
We could check if it changed, like:
```
When("I go to previous step", () => {
 const initialUrl
  cy.get("[data-test='wizard-previous-button']").click();
const finalUrl

test initialUrl === finalUrl
})
```

Is that what you mean?

Scenarions that use this step, check the url in the following step:

```
  Scenario: Go back to deal type select
    Given I am presented with Open Proposal proposal stage
    When I go to previous step
    Then I am presented the option to choose a partner
```

I wanted to make that step generic

Reply:
I don't know how to check if url changed, because in cypress we cannot variables in the way we do in normal JS. If you know the solution please suggest it

Reply:
> generic

True.
Though I don't see verifying "url changed" against "being generic".

If I read this Scenario from a Engineer perspective
```feature
Scenario: Go back to deal type select 
  Given I am presented with Open Proposal proposal stage 
  When I go to previous step 
  Then I am presented the option to choose a partner
```

Then in the `When` step, is where I would have put the "generic url test".

---
Also, that wouldn't be a problem, if multiple steps test for the url.
If Steps are generic they should cover a bit rather than a bit less imo.

Reply:
> suggestions

Can you check if this works
```ts
cy.url().then(url => {
  const oldUrl = url;
  // do click
  cy.get().click()

  // verify url change
  cy.url().should('equal', oldUrl);
})
```

Reply:
For sure, I completely agree that we should check if url changed, sorry for misunderstanding. 

Thanks for the code snippet, it works!